src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -1186,9 +1186,9 @@ static void get_max_dev_id(uint32_t *max_dev_id, uint32_t *max_subdev_id)
 
     /* Values based on ZE_AFFINITY_MASK */
     for (int i = 0; i < mask_contents.num_dev; ++i) {
-        if (mask_contents.dev_id[i] > *max_dev_id)
+        if (mask_contents.dev_id[i] > (int) *max_dev_id)
```

Comment:
Wait, the issue is `max_dev_id` need go to `-1` and it is wrongly typed to `uint32_t`, and the solution is to cast? Shouldn't we fix the parameter types in `get_max_dev_id` (and fix the types in calling sites as well)?

Reply:
No the problem is that `mask_contents.dev_id[i]` can be -1, and because `max_dev_id` is unsigned, the comparison is treated as unsigned, so it incorrectly evaluates the branch as true (i.e. `UINT_MAX > 0`). I tested this independently:

```
#include <stdio.h>
#include <stdint.h>

int main(int argc, char **argv) {
    int dev_id = -1;
    uint32_t max_dev_id = 0;

    if (dev_id > max_dev_id) {
        max_dev_id = dev_id;
    }
    fprintf(stderr, "%lu\n", max_dev_id);

    dev_id = -1;
    max_dev_id = 0;

    if (dev_id > (int) max_dev_id) {
        max_dev_id = dev_id;
    }
    fprintf(stderr, "%lu\n", max_dev_id);
}
```

output:

```
> gcc -g -O0 -o convert convert.c
> ./convert
4294967295
0
```

Reply:
To your original suggestion though, I would be okay to instead change everything to `signed` (including the types)

Reply:
Reverted back to `int` parameters for `get_max_dev_id` and changed `max_dev_id` and `max_subdev_id` to `int`

Reply:
Thank you for the explanation. Yeah, I think using `int` is more robust.